### DISCUSSION TITLE
How do I set a breakpoint in dependency code?

### DISCUSSION THREAD

**Discussion by dboreham** [scalameta/metals#3067] [General]
I want to debug into a dependency. I've been able to make this work by setting a breakpoint in my application code in a place where it gets called back from some place in the dependency. Then by clicking up the stack trace displayed in VSCode I can get metals to open the relevant dependency code at `.metals/readonly/dependencies\<jar-file>\...`. I can set additional breakpoints in that same source file since it is open on the screen. So far so good. However, I can't see how to set a breakpoint in a "sibling" dependency source file that hasn't been "magically" copied into the readonly hierarchy by clicking on a function in a stack trace. The only files there appear to be files that have already been opened by the stack trace clicking trick. Is there a secret to how to say "open a file in that jar please" so I can set breakpoints in it?

**Comment by tgodzik** [comment]
Thanks for the question! It used to be possible to go and search library sources by adding `;` at the end when searching for symbol in workspace. Unfortunately, this is currently being filtered by VS Code. It's still possible to navigate to sibling source by going into the metals view then searching for the appropriate library in the `packages` top window.

![image](https://user-images.githubusercontent.com/3807253/130810383-185aa640-b610-4868-8daa-2e0fadea3d68.png)

**Reply by ches** [reply]
At this time browsing Scala 3 dependencies in the package browser this way isn't working, see #2859
